src/term/net_to_term.rs

```diff
@@ -211,6 +214,42 @@ impl<'a> Reader<'a> {
 
     term
   }
+
+  /// Enters both ports 1 and 2 of a node  
+  /// If both ports are connected to the same node, and that node Kind is the same as the given node Kind,  
+  /// Reads the port 0 of the connected node, and returns that term,  
+  /// Otherwise, returns the terms on ports 1 and 2 of the given node.
+  ///
+  /// # Example
+  ///
+  ///
+  /// ```text
+  /// // λa let (a, b) = a; (a, b)
+  /// ([a b] [a b])
+  /// 
+  /// The node `(a, b)` is just a reconstruction of the destructuring of `a`,  
+  /// So we can skip both steps and just return the "value" unchanged:
+  /// 
+  /// // λa a
+  /// (a a)
+  /// ```
+  ///
+  fn decay_or_get_ports(&mut self, node: NodeId) -> Result<Term, (Term, Term)> {
+    let fst_port = self.net.enter_port(Port(node, 1));
+    let snd_port = self.net.enter_port(Port(node, 2));
+
+    if fst_port.node() == snd_port.node() && fst_port.slot() == 1 && snd_port.slot() == 2 {
+      let first_node = fst_port.node();
+      if self.net.node(node).kind == self.net.node(first_node).kind {
```

Comment:
This is not valid for all kinds of nodes, only CON/TUP/DUP, due to  their interaction rules.